src/server/host/Endpoints/Events/PatchPlayerEventRegistrationsEndpoint.cs

```diff
@@ -0,0 +1,190 @@
+            return;
+        }
+
+        var registrations = await databaseContext
+            .EventTimeslotRegistrations.Where(x =>
+                x.Player.Id == userId && x.EventTimeslot.EventId == eventId
+            )
+            .ToArrayAsync(ct);
+        var targetRegistration = new
+        {
+            TimeslotId = idService.EventTimeslot.DecodeSingle(req.TimeslotId)
+        };
+        var timeslotToModify = registrations.FirstOrDefault(x =>
+            x.EventTimeslotId == targetRegistration.TimeslotId
+        );
+        if (timeslotToModify == null && req.IsRegistered)
+        {
+            // not existent but wants to -> then add
+            databaseContext.EventTimeslotRegistrations.Add(
+                new EventTimeslotRegistrationEntity
+                {
+                    EventTimeslot = databaseContext.EventTimeslotById(
+                        targetRegistration.TimeslotId
+                    ),
+                    Player = databaseContext.UserById(userId),
+                    FallbackEventTimeslot = null
+                }
+            );
+        }
+        else if (timeslotToModify != null && !req.IsRegistered)
+        {
+            // is existent but does not want to -> remove
+            databaseContext.EventTimeslotRegistrations.RemoveRange(
+                registrations.Where(x => targetRegistration.TimeslotId == x.EventTimeslotId)
+            );
+        }
+        await databaseContext.SaveChangesAsync(ct);
+        await SendAsync(null, cancellation: ct);
+
+        await realtimeEventsService.SendEventAsync(
+            new RealtimeEvent.PlayerEventRegistrationChanged(
+                idService.User.Encode(userId),
+                idService.Event.Encode(eventId)
+            ),
+            ct
+        );
+        var userAlias = await databaseContext
+            .Users.Where(x => x.Id == userId)
+            .Select(x => x.Alias)
+            .FirstOrDefaultAsync(ct);
+        var changeEvents = registrations
+            .Where(x => targetRegistration.TimeslotId == x.EventTimeslotId)
+            .Select(x => new RealtimeEvent.PlayerEventTimeslotRegistrationChanged(
+                idService.Event.Encode(eventId),
+                idService.EventTimeslot.Encode(x.EventTimeslotId),
+                idService.User.Encode(userId),
+                userAlias,
+                req.IsRegistered
+            ))
+            .Concat(
+                new RealtimeEvent.PlayerEventTimeslotRegistrationChanged(
+                    idService.Event.Encode(eventId),
+                    idService.EventTimeslot.Encode(targetRegistration.TimeslotId),
+                    idService.User.Encode(userId),
+                    userAlias,
+                    req.IsRegistered
+                )
+            );
+        foreach (var changeEvent in changeEvents)
+            await realtimeEventsService.SendEventAsync(changeEvent, ct);
```

Comment:
```suggestion
        await realtimeEventsService.SendEventAsync(
            new RealtimeEvent.PlayerEventTimeslotRegistrationChanged(
                idService.Event.Encode(eventId),
                idService.EventTimeslot.Encode(targetRegistration.TimeslotId),
                idService.User.Encode(userId),
                userAlias,
                req.IsRegistered
            ) ,
            ct
        );
```

This should be enough, right?